feat(test-runner):include test step in json report

keevie opened the pull request:
references https://github.com/microsoft/playwright/issues/8475

keevie committed:
feat(test-runner):include test step in json report

keevie committed:
feat(test-runner): include test step in json report

Ensure that we don't have empty steps lists; add some tests

dgozman reviewed (commented):
This looks very good apart from the filter.

keevie committed:
feat(test runner): include step in json report

don't include useless category information (it will always be test.step)
rm console log

Ni55aN commented:
https://github.com/microsoft/playwright/blob/4b0e55abda3be870f17f35c6dc5db92db61c44d4/packages/playwright/src/reporters/json.ts#L192

I would be nice to have an options include any category.

Besides thay, in my test cases I don’t find a any step where the category === `test.step`.  Usually it contains `pw:api`,

Ni55aN commented:
https://github.com/microsoft/playwright/blob/4b0e55abda3be870f17f35c6dc5db92db61c44d4/packages/playwright/src/reporters/json.ts#L192

It would be nice to have an options include any category
